runtime/src/main/cpp/Memory.cpp

```diff
@@ -1759,7 +1761,17 @@ void GarbageCollect() {
 
   incrementStack(state);
   processDecrements(state);
+  size_t beforeDecrements = state->toRelease->size();
   decrementStack(state);
+  size_t afterDecrements = state->toRelease->size();
+  ssize_t stackReferences = afterDecrements - beforeDecrements;
+  if (stackReferences * 5 > state->gcThreshold) {
+     auto newThreshold = state->gcThreshold * 3 / 2 + 1;
+     if (newThreshold < kMaxErgonomicThreshold) {
+       state->gcThreshold = newThreshold;
```

Comment:
Is space in `toRelease` not reserved intentionally?

Reply:
Reworked here.